src/main/java/seedu/address/logic/commands/AddRestaurantNoteCommand.java

```diff
@@ -63,61 +65,46 @@ public AddRestaurantNoteCommand(Index index, Note recommendedFood, Note goodFood
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
         List<Restaurant> lastShownList = model.getFilteredRestaurantList();
 
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_RESTAURANT_DISPLAYED_INDEX);
         }
 
         Restaurant restaurantToEdit = lastShownList.get(index.getZeroBased());
-
-        ArrayList<Note> updatedRecFood = new ArrayList<>();
-        for (Note i : restaurantToEdit.getRecommendedFood()) {
-            updatedRecFood.add(i);
-        }
-
-        ArrayList<Note> updatedGoodFood = new ArrayList<>();
-        for (Note i : restaurantToEdit.getGoodFood()) {
-            updatedGoodFood.add(i);
-        }
-
-        ArrayList<Note> updatedBadFood = new ArrayList<>();
-        for (Note i : restaurantToEdit.getBadFood()) {
-            updatedBadFood.add(i);
+        if (recommendedFood.size() == 0 || goodFood.size() == 0 || badFood.size() == 0) {
+            throw new CommandException(MESSAGE_EMPTY);
         }
 
-        if (!recommendedFood.note.isEmpty()) {
-            updatedRecFood.add(recommendedFood);
+        if (recommendedFood.size() != 0) {
+            for (int i = 0; i < recommendedFood.size(); i++) {
+                restaurantToEdit.getRecommendedFood().add(recommendedFood.get(i));
+            }
         }
-        if (!goodFood.note.isEmpty()) {
-            updatedGoodFood.add(goodFood);
+        if (goodFood.size() != 0) {
+            for (int j = 0; j < goodFood.size(); j++) {
+                restaurantToEdit.getGoodFood().add(goodFood.get(j));
+            }
         }
-        if (!badFood.note.isEmpty()) {
-            updatedBadFood.add(badFood);
+        if (badFood.size() != 0) {
+            for (int k = 0; k < badFood.size(); k++) {
+                restaurantToEdit.getBadFood().add(badFood.get(k));
+            }
```

Comment:
Not sure if this is allowed maybe you gotta do it this way:

ArrayList<Note> updatedList = restaurant.getBadFood()
Add the new notes to this updatedList and then create a new EditedRestaurant?